server/src/main/java/org/elasticsearch/index/mapper/PassThroughObjectMapper.java

```diff
@@ -14,26 +14,37 @@
 import org.elasticsearch.xcontent.XContentBuilder;
 
 import java.io.IOException;
+import java.util.Collection;
+import java.util.HashMap;
 import java.util.Locale;
 import java.util.Map;
 
 import static org.elasticsearch.common.xcontent.support.XContentMapValues.nodeBooleanValue;
+import static org.elasticsearch.common.xcontent.support.XContentMapValues.nodeIntegerValue;
 
 /**
  * Mapper for pass-through objects.
  *
  * Pass-through objects allow creating fields inside them that can also be referenced directly in search queries.
  * They also include parameters that affect how nested fields get configured. For instance, if parameter "time_series_dimension"
  * is set, eligible subfields are marked as dimensions and keyword fields are additionally included in routing and tsid calculations.
+ *
+ * In case different pass-through objects contain subfields with the same name (excluding the pass-through prefix), their aliases conflict.
+ * To resolve this, the pass-through spec specifies which object takes precedence through required parameter "priority"; non-negative
+ * integer values are accepted, with the highest priority value winning in case of conflicting aliases.
```

Comment:
Priority 1 is the highest priority, right?

Reply:
No 0 is the lowest. I can change it if you feel strongly, do we have another case where we use priority in a field?

Reply:
Oh, I see. No strong opinion here, mainly wanted to understand whether priority is ascending or descending. Maybe a slight preference for only allowing positive integers (excluding zero).

>  do we have another case where we use priority in a field?

I've found 
https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/common/Priority.java. Interesting it does have a priority 0 (`IMMEDIATE`). But that seems to be a bit of a different use case.

I've also found we're using `priority` for [index templates](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-templates.html) and the higher the number, the higher the priority, so priority 1 would be a very low priority.

Reply:
Right, I think Martijn suggested to use the same approach as the last example.